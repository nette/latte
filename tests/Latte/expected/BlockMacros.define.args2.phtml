<?php
%A%

final class Template%a% extends Latte\Runtime\Template
{
	protected const BLOCKS = [
		['test' => 'blockTest'],
	];


	public function main(): array
	{
		extract($this->params);
		echo 'named arguments

';
		if ($this->getParentName()) {
			return get_defined_vars();
		}
		echo '
a) ';
		$this->renderBlock($__nm = 'test', [1, 'var1' => 2] + [], 'html');
		echo '

b) ';
		$this->renderBlock($__nm = 'test', ['var2' => 1] + [], 'html');
		echo '

c) ';
		$this->renderBlock($__nm = 'test', ['hello' => 1] + [], 'html');
		echo '

d) ';
		$this->renderBlock($__nm = 'test', ['var2' => 1, 2] + [], 'html');
		echo ' // invalid';
		return get_defined_vars();
	}


	public function blockTest(array $__args): void
	{
		extract($this->params);
		$var1 = $__args[0] ?? $__args['var1'] ?? null;
		$var2 = $__args[1] ?? $__args['var2'] ?? null;
		$var3 = $__args[2] ?? $__args['var3'] ?? null;
		echo '	Variables ';
		echo LR\Filters::escapeHtmlText($var1) /* line 4 */;
		echo ', ';
		echo LR\Filters::escapeHtmlText($var2) /* line 4 */;
		echo ', ';
		echo LR\Filters::escapeHtmlText($var3) /* line 4 */;
		echo "\n";
	}

}
