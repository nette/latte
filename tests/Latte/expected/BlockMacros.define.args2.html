named arguments


a) 	Variables 1, ,


b) 	Variables , 1,


c) 	Variables , ,


d) 	Variables 2, 1,
 // invalid
